src/Stan/Analysis/Analyser.hs

```diff
@@ -83,6 +84,23 @@ analyseBigTuples insId = matchAstWith isBigTuple insId tuple
         _:_:_:_:_  -> True
         _lessThan4 -> False
 
+{- | Check for usage of the 'String' type in the following places:
+
+* Type signatures: foo :: a -> String
+* Data type fields: newtype HasString = HasString { foo :: String }
+-}
+analyseStringUsage
+    :: Id Inspection
+    -> HieFile
+    -> HieAST TypeIndex
+    -> State VisitorState ()
+analyseStringUsage insId = matchAstWith usesString insId typeSig
+  where
+    usesString :: HieAST TypeIndex -> Bool
+    usesString = isJust . hieFindIdentifier stringName
```

Comment:
Apparently this function `hieFindIdentifier` is not doing what I expected 😕 .
There is no other usage in the code, maybe I'm using it wrong? I've confirmed that what I'm looking for are `nodeIdentifiers` that contain my given `NameMeta`...
A little more help on this would be greatly appreciated 🙏  @chshersh @vrom911